include/deal.II/base/vectorization.h

```diff
@@ -1592,6 +1698,68 @@ vectorized_load_and_transpose(const unsigned int          n_entries,
 
 
 
+/**
+ * Specialization for float and AVX-512.
+ */
+template <>
+inline DEAL_II_ALWAYS_INLINE void
+vectorized_load_and_transpose(const unsigned int             n_entries,
+                              const std::array<float *, 16> &in,
+                              VectorizedArray<float, 16> *   out)
+{
+  // Similar to the double case, we perform the work on smaller entities. In
+  // this case, we start from 128 bit arrays and insert them into a full 512
+  // bit index. This reduces the code size and register pressure because we do
+  // shuffles on 4 numbers rather than 16.
+  const unsigned int n_chunks = n_entries / 4;
+
+  // To avoid warnings about uninitialized variables, need to initialize one
+  // variable to a pre-exisiting value in out, which will never get used in
+  // the end. Keep the initialization outside the loop because of a bug in
+  // gcc-9 which generates a "vmovapd" instruction instead of "vmovupd" in
+  // case t3 is initialized to zero (inside/outside of loop), see
+  // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90991
```

Comment:
I would remove these comments here and refer to the other code. I would instead add a comment (maybe in the central templated variant) why we have two different functions and why we cannot simply let one function call the other - we either have to increment only the base pointer in case of a single offset, but we need to increment all pointers like in the new case. And we want the compiler to be able to perform one or the other. To be honest, sometimes the compiler transforms the case with a single base with n offsets into n separate pointers that are all incremented (especially when we only do 4 updates), but for wider SIMD the increment of the base is better.